### PR TITLE
Operator docs for Source.combine

### PR DESCRIPTION
**Pull request by johanandren**
Refs #25468

### CODE REVIEW

File: akka-docs/src/main/paradox/stream/operators/Source/combine.md [hunk 1]
```diff
@@ -14,22 +14,56 @@ Combine several sources, using a given strategy such as merge or concat, into on
 
 ## Description
 
-## Reactive Streams semantics
-
-@@@div { .callout }
+Provides a way to create a "fan-in" of multiple sources without having to use the more advanced @ref:[GraphDSL](../../stream-graphs.md#constructing-graphs).
+
+The way the elements from the sources
+is combined is pluggable through the `strategy` parameter which accepts a function
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/Source/combine.md]:
```suggestion
are combined is pluggable through the `strategy` parameter which accepts a function 
```

File: akka-docs/src/main/paradox/stream/operators/Source/combine.md [hunk 2]
```diff
@@ -14,22 +14,56 @@ Combine several sources, using a given strategy such as merge or concat, into on
...
 
-## Reactive Streams semantics
-
-@@@div { .callout }
+Provides a way to create a "fan-in" of multiple sources without having to use the more advanced @ref:[GraphDSL](../../stream-graphs.md#constructing-graphs).
+
+The way the elements from the sources
+is combined is pluggable through the `strategy` parameter which accepts a function 
+@scala[`Int => Graph[FanInShape]`]@java[`Integer -> Graph<FanInShape>`] where the integer parameter specifies the number of sources
+that the graph must accept. This makes it possible to use `combine` it with the built in `Concat`
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/Source/combine.md]:
```suggestion
that the graph must accept. This makes it possible to use `combine` with the built-in `Concat` 
```

File: akka-docs/src/main/paradox/stream/operators/Source/combine.md [hunk 3]
```diff
@@ -14,22 +14,56 @@ Combine several sources, using a given strategy such as merge or concat, into on
...
-## Reactive Streams semantics
-
-@@@div { .callout }
+Provides a way to create a "fan-in" of multiple sources without having to use the more advanced @ref:[GraphDSL](../../stream-graphs.md#constructing-graphs).
+
+The way the elements from the sources
+is combined is pluggable through the `strategy` parameter which accepts a function 
+@scala[`Int => Graph[FanInShape]`]@java[`Integer -> Graph<FanInShape>`] where the integer parameter specifies the number of sources
+that the graph must accept. This makes it possible to use `combine` it with the built in `Concat` 
+and `Merge` by @scala[expanding their `apply` methods to functions]@java[using a method reference to their `create` methods]
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/Source/combine.md]:
```suggestion
and `Merge` by @scala[expanding their `apply` methods to functions]@java[using a method reference to their `create` methods],
```

File: akka-docs/src/main/paradox/stream/operators/Source/combine.md [hunk 4]
```diff
@@ -14,22 +14,56 @@ Combine several sources, using a given strategy such as merge or concat, into on
...
+Provides a way to create a "fan-in" of multiple sources without having to use the more advanced @ref:[GraphDSL](../../stream-graphs.md#constructing-graphs).
+
+The way the elements from the sources
+is combined is pluggable through the `strategy` parameter which accepts a function 
+@scala[`Int => Graph[FanInShape]`]@java[`Integer -> Graph<FanInShape>`] where the integer parameter specifies the number of sources
+that the graph must accept. This makes it possible to use `combine` it with the built in `Concat` 
+and `Merge` by @scala[expanding their `apply` methods to functions]@java[using a method reference to their `create` methods] 
+but also to use an arbitrary strategy. 
+
+Combine is most useful when you have more sources than 2 or want to use a custom operator since there are more concise
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/Source/combine.md]:
```suggestion
Combine is most useful when you have more sources than 2 or want to use a custom operator, as there are more concise 
```

File: akka-docs/src/main/paradox/stream/operators/Source/combine.md [hunk 5]
```diff
@@ -14,22 +14,56 @@ Combine several sources, using a given strategy such as merge or concat, into on
...
+is combined is pluggable through the `strategy` parameter which accepts a function 
+@scala[`Int => Graph[FanInShape]`]@java[`Integer -> Graph<FanInShape>`] where the integer parameter specifies the number of sources
+that the graph must accept. This makes it possible to use `combine` it with the built in `Concat` 
+and `Merge` by @scala[expanding their `apply` methods to functions]@java[using a method reference to their `create` methods] 
+but also to use an arbitrary strategy. 
+
+Combine is most useful when you have more sources than 2 or want to use a custom operator since there are more concise 
+operators for 2-source @ref:[concat](../Source-or-Flow/concat.md) and @ref:[merge](../Source-or-Flow/merge.md) 
+
+Some of the built in operators that can be used as strategy are:
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/Source/combine.md]:
```suggestion
Some of the built-in operators that can be used as strategy are:
```

File: akka-docs/src/main/paradox/stream/operators/Source/combine.md [hunk 6]
```diff
@@ -14,22 +14,56 @@ Combine several sources, using a given strategy such as merge or concat, into on
...
+ 
+ * @apidoc[akka.stream.*.Merge] 
+ * @apidoc[akka.stream.(javadsl|scaladsl).Concat]
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/Source/combine.md]: I forgot this was a thing, neat

**Review by raboof — APPROVED** [review]
Very nice contrasting these 2 different strategies like that.

Sprinkled some typography, see if you like it, feel free to ignore if you don't ;).

File: akka-docs/src/main/paradox/stream/operators/Source/combine.md (hunk 6 above)
Comment (johanandren) [reply akka-docs/src/main/paradox/stream/operators/Source/combine.md]: Yeah, we have a third in `akka.stream.internal.Concat` so `akka.stream.*.Concat` gave too many options.

**Comment by johanandren** [comment]
PLS BUILD

File: akka-docs/src/test/scala/docs/stream/operators/source/Combine.scala [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  @throws[Exception]
+  def concat(): Unit = {
+    // #source-combine-concat
+    val source1 = Source(1 to 3)
+    val source2 = Source(8 to 10)
+    val source3 = Source(12 to 14)
+    val sources = Source.combine(source1, source2, source3)(Concat(_))
+    sources.runForeach(println)
+    // prints (order is deterministic)
+    // #source-combine-concat
```
Comment (ennru) [inline akka-docs/src/test/scala/docs/stream/operators/source/Combine.scala]: The printout is missing.

File: akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala [hunk 8]
```diff
@@ -731,7 +731,7 @@ object Source {
     }, { case akka.actor.Status.Failure(cause)              => cause })
 
   /**
-   * Combines several sources with fan-in strategy like `Merge` or `Concat` and returns `Source`.
+   * Combines several sources with fan-in strategy like [[Merge]] or [[Concat]] into a single [[Source]].
```
Comment (ennru) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala]: Apply this to `javadsl` and to `combineMat`, as well?

**Review by ennru — APPROVED** [review]
Great improvement!

**Pull request merged**